replicaset.md

```diff
@@ -0,0 +1,190 @@
+﻿## Virtual Machine Replica Set
+
+A *VirtualMachineReplicaSet* tries to ensures that a specified number of
+Virtual Machine replicas are running at any time. In other words, a
+*VirtualMachineReplicaSet* makes sure that a Virtual Machine or a homogeneous set of
+Virtual Machines is always up and ready. It is very similar to a 
+[Kubernetes Replica Set](https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/).
+
+No state is kept and no guarantees about the maximum number of Virtual Machine
+Replicas which are up are given. For example, the *VirtualMachineReplicaSet*
+may decide to create new Replicas if possibly still running Virtual Machines
+are entering an unknown state. 
+
+### How to use a VirtualMachineReplicaSet
+
+The *VirtualMachineReplicaSet* allows to specify a *VirtualMachineTemplate* in
+`spec.template`. It consists of `ObjectMetadata` in `spec.template.metadata`, and
+a `VirtualMachineSpec` in `spec.template.spec`. The specification of the virtual
+machine is equal to the specification of the virtual machine in the
+`VirtualMachine` workload.
+
+`spec.replicas` can be used to specify how much replicas are wanted. If not
+specified, the default value is 1. This value can be updated anytime. The
+controller will react to the changes.
+
+`spec.selector` is used by the controller to keep track of managed virtual
+machines. The selector specified there must be able to match the virtual
+machine labels as specified in `spec.template.metadata.labels`. If the selector
```

Comment:
This is a little strange from a user perspective. I would expect the VMRS to manage this label for us unless the user explicitly defines the label themselves.

Reply:
Yes it is not that nice. We can improve it a little by doing defaulting in both ways if possible. With the set based matchers it is not always possible to deduce labels for the VM.